### PR TITLE
New Tab Page: Prompt for Brave Together 

### PR DESCRIPTION
**Pull request by petemill**
- Button in footer always shows
- Prompt (anchored to button) shows like a tooltip until the user manually closes via button.
- _(Also adds dir="rtl" as attribute in instead of stylesheet so that it's targetable as a css selector. This is done in most chromium webui now.)_

Resolves https://github.com/brave/brave-browser/issues/12776

image

image

### Button Transitions

button-transitions

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import styled from 'brave-ui/theme'
+
+const Button = styled('button')`
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx:8]: Apparently this is a new design system-able button style that we can make more shared when the design team iterates a bit more

File: components/brave_new_tab_ui/reducers/new_tab_reducer.ts [hunk 2]
```diff
@@ -186,6 +187,14 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       break
     }
 
+    case Actions.dismissTogetherPrompt.getType(): {
```
Comment (zenparsing) [inline components/brave_new_tab_ui/reducers/new_tab_reducer.ts:190]: It looks like you are preferring not to add an exported constant to `new_tab_types`. Do you want to eventually remove the need for that file completely?

File: components/brave_new_tab_ui/components/default/footer/togetherTooltip/style.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    on the original and clip the copy to only the "arrow". */
+  &:before {
+    content: " ";
+    position: absolute;
+    top: 0;
+    bottom: calc(-1 * var(--arrow-width-half));
+    left: 0;
+    right: 0;
+    background: linear-gradient(305.95deg, #BF14A2 0%, #F73A1C 98.59%);
+    clip-path: polygon(
```
Comment (zenparsing) [inline components/brave_new_tab_ui/components/default/footer/togetherTooltip/style.ts:47]: Neat!

**Review by zenparsing — APPROVED** [review]

File: components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx:1]: oops, it's 2021 now

File: components/brave_new_tab_ui/reducers/new_tab_reducer.ts (hunk 2 above)
Comment (petemill) [reply components/brave_new_tab_ui/reducers/new_tab_reducer.ts:190]: Yes just trying to remove 1 layer of boilerplate. I've been using `redux-act` for Brave Today on the NTP via `today_actions.ts` and `reducers/today`, so just continuing that here for anything new.

File: components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx (hunk 4 above)
Comment (petemill) [reply components/brave_new_tab_ui/components/default/footer/togetherTooltip/button.tsx]: fixed

**Comment by petemill** [comment]
Rebased in order to try to fix CI sync errors. Also added a change to storybook in order to provide an option for right-to-left so that we can easily preview what the browser does on the NTP for certain languages. https://github.com/brave/brave-core/pull/7532/commits/d880de20c1511e89c63bf8c5384fb3807f1a90aa

**Comment by karenkliu** [comment]
@petemill The icon in the tooltip prompt should appear on the right side before the title string for right-to-left languages:

Frame 1

**Pull request merged**